tests/filecheck/backend/riscv/canonicalize.mlir

```diff
@@ -139,8 +146,13 @@ builtin.module {
 // CHECK-NEXT:   %load_float_known_offset = riscv.flw %i2, 12 : (!riscv.reg<>) -> !riscv.freg<fa0>
 // CHECK-NEXT:   "test.op"(%load_float_known_offset) : (!riscv.freg<fa0>) -> ()
 
+// CHECK-NEXT:   %load_double_known_offset = riscv.fld %i2, 12 : (!riscv.reg<>) -> !riscv.freg<fa0>
+// CHECK-NEXT:   "test.op"(%load_double_known_offset) : (!riscv.freg<fa0>) -> ()
+
 // CHECK-NEXT:   riscv.fsw %i2, %f2, 12 : (!riscv.reg<>, !riscv.freg<>) -> ()
 
+// CHECK-NEXT:   riscv.fsd %i2, %f2, 12 : (!riscv.reg<>, !riscv.freg<>) -> ()
```

Comment:
Just out of curiosity: where is addi removed?

Reply:
It's removed by dce at the end of the canonicalization pass